src/query/api/v1/middleware/rewrite.go

```diff
@@ -263,3 +334,44 @@ func maybeUpdateLookback(
 	}
 	return false, lookback
 }
+
+type fakeQueryable struct {
+	engine              *promql.Engine
+	instant             bool
+	calculatedStartTime time.Time
+	calculatedEndTime   time.Time
+}
+
+func (f *fakeQueryable) Querier(ctx context.Context, mint, maxt int64) (promstorage.Querier, error) {
+	f.calculatedStartTime = xtime.FromUnixMillis(mint)
+	f.calculatedEndTime = xtime.FromUnixMillis(maxt)
+	// fail here to cause prometheus to give up on query execution
+	return nil, fmt.Errorf("ignorable error")
```

Comment:
Should define this error on top of file and use it.. Makes the comparison better..

Reply:
Good idea. Done